src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -4984,6 +4985,38 @@ private SourceLocation getSelectionAsSourceLocation(boolean fromClick)
       int column = selPos.getColumn() + 1;
       return SourceLocation.create(file, line, column, fromClick);
    }
+   
+   @Handler
+   void onQuickAddNext()
+   {
+      if (!(docDisplay_ instanceof AceEditor))
```

Comment:
Ah, no, it makes sense -- I just had a bit of a mental block on how to restructure the code (ie, literally just move the implementation to `AceEditor`; don't try to expose more guts). Thanks for the explanation!